### PR TITLE
Add github action CI, to generate a fetchable executable for other projects

### PR DESCRIPTION
**Pull request by fredroy**
The GA is triggered only on demand, as we dont really care to regenerate the exe with new regression references files

### CODE REVIEW

File: Regression_test/CMakeLists.txt [hunk 1]
```diff
@@ -22,3 +22,10 @@ add_executable(${PROJECT_NAME} ${SOURCE_FILES} ${HEADER_FILES})
 add_definitions("-DSOFA_SRC_DIR=\"${CMAKE_SOURCE_DIR}\"")
 target_link_libraries(${PROJECT_NAME} PUBLIC Sofa.Testing)
 target_link_libraries(${PROJECT_NAME} PUBLIC Sofa.Component)
+
+sofa_add_targets_to_package(
+    PACKAGE_NAME SofaGui
+    TARGETS ${PROJECT_NAME} AUTO_SET_TARGET_PROPERTIES
+    INCLUDE_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}"
+    INCLUDE_INSTALL_DIR "SofaGui/${PROJECT_NAME}"
+    )
```
Comment (guparan) [inline Regression_test/CMakeLists.txt:31]: Why under SofaGui ?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+          path: artifacts
+
+      - name: Zip artifacts
+        shell: bash
+        run: |
+          cd $GITHUB_WORKSPACE/artifacts
+          for artifact in *; do
+            zip $artifact.zip -r $artifact/*
+          done
+      - name: Upload release
```
Comment (guparan) [inline .github/workflows/ci.yml:92]:
```suggestion
          done
      
      - name: Upload release
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        uses: actions/upload-artifact@v2
+        with:
+          name: Regression_test_${{ steps.sofa.outputs.run_branch }}_for-SOFA-${{ steps.sofa.outputs.sofa_version }}_${{ runner.os }}
+          path: ${{ env.WORKSPACE_INSTALL_PATH }}
+
+      - name: Install artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: Regression_test_${{ steps.sofa.outputs.run_branch }}_for-SOFA-${{ steps.sofa.outputs.sofa_version }}_${{ runner.os }}
+          path: ${{ env.WORKSPACE_ARTIFACT_PATH }}
```
Comment (guparan) [inline .github/workflows/ci.yml:71]: This step is not needed.
```suggestion
```

**Review by guparan — CHANGES_REQUESTED** [review]

File: Regression_test/CMakeLists.txt (hunk 1 above)
Comment (fredroy) [reply Regression_test/CMakeLists.txt:31]: I copied-paste from runSofa 🙊 which has this package name as well... what should we set ?

**Pull request merged**